app/uk/gov/hmrc/ngrraldfrontend/controllers/ProvideDetailsOfFirstSecondRentPeriodController.scala

```diff
@@ -0,0 +1,201 @@
+        classes = "govuk-fieldset__legend--s",
+        isPageHeading = true
+      ))
+    )),
+    hint = Some(Hint(
+      id = Some("provideDetailsOfFirstSecondRentPeriod.firstPeriod.start.date.hint"),
+      content = Text(messages("provideDetailsOfFirstSecondRentPeriod.firstPeriod.start.date.hint"))
+    ))
+  )
+
+  def firstDateEndInput()(implicit messages: Messages): DateInput = DateInput(
+    id = "provideDetailsOfFirstSecondRentPeriod.firstPeriod.end.date",
+    namePrefix = Some("provideDetailsOfFirstSecondRentPeriod.firstPeriod.end.date"),
+    fieldset = Some(Fieldset(
+      legend = Some(Legend(
+        content = Text(messages("provideDetailsOfFirstSecondRentPeriod.firstPeriod.end.date.label")),
+        classes = "govuk-fieldset__legend--s",
+        isPageHeading = true
+      ))
+    )),
+    hint = Some(Hint(
+      id = Some("provideDetailsOfFirstSecondRentPeriod.firstPeriod.end.date.hint"),
+      content = Text(messages("provideDetailsOfFirstSecondRentPeriod.firstPeriod.end.date.hint"))
+    ))
+  )
+
+  private def firstRentPeriodYesButton(form: Form[ProvideDetailsOfFirstSecondRentPeriodForm])(implicit messages: Messages): NGRRadioButtons = NGRRadioButtons(
+    radioContent = "provideDetailsOfFirstSecondRentPeriod.firstPeriod.radio.yes",
+    radioValue = yesPayedRent,
+    conditionalHtml = Some(inputText(
+      form = form,
+      id = "RentPeriodAmount",
+      name = "RentPeriodAmount",
+      label = messages("provideDetailsOfFirstSecondRentPeriod.firstPeriod.radio.conditional.hint.bold"),
+      isVisible = true,
+      hint = Some(Hint(
+        content = Text(messages("provideDetailsOfFirstSecondRentPeriod.firstPeriod.radio.conditional.hint"))
+      )),
+      classes = Some("govuk-input--width-10"),
+      prefix = Some(PrefixOrSuffix(content = Text("£")))
+    ))
+  )
+
+  private def firstRentPeriodRadio(form: Form[ProvideDetailsOfFirstSecondRentPeriodForm])(implicit messages: Messages): NGRRadio = {
+    val ngrRadioButtons: Seq[NGRRadioButtons] = Seq(
+      firstRentPeriodYesButton(form),
+      NGRRadioButtons(radioContent = "provideDetailsOfFirstSecondRentPeriod.firstPeriod.radio.no", radioValue = noRentPayed),
+    )
+    NGRRadio(
+      NGRRadioName("provideDetailsOfFirstSecondRentPeriod-radio-firstRentPeriodRadio"),
+      ngrTitle = Some(Legend(content = Text(messages("provideDetailsOfFirstSecondRentPeriod.firstPeriod.radio.label")), classes = "govuk-fieldset__legend--m", isPageHeading = true)),
+      NGRRadioButtons = ngrRadioButtons
+    )
+  }
+
+  def secondDateStartInput()(implicit messages: Messages): DateInput = DateInput(
+    id = "provideDetailsOfFirstSecondRentPeriod.secondPeriod.start.date",
+    namePrefix = Some("provideDetailsOfFirstSecondRentPeriod.secondPeriod.start.date"),
+    fieldset = Some(Fieldset(
+      legend = Some(Legend(
+        content = Text(messages("provideDetailsOfFirstSecondRentPeriod.secondPeriod.start.date.label")),
```

Comment:
Could you add this to the messages file please

Reply:
You will also need to provide a message for this error as well provideDetailsOfFirstSecondRentPeriod.second.startDate.required.error

Reply:
If I click Yes in the first rent period to the Do you pay rent in this period, an error doesn't show . It shows all the other errors but nothing for the currency input